runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -306,6 +310,33 @@ local function get_spell(capture_name)
   return nil, 0
 end
 
+---@param m MarkInfo
+---@param buf integer
+---@param line integer
+---@param next_marks MarkInfo[]
+local function add_mark(m, buf, line, next_marks)
```

Comment:
is there a 1-line docstring that can give insight into what this does? E.g. "Sets a mark unless..."